librz/demangler/demangler.c

```diff
@@ -0,0 +1,184 @@
+// SPDX-FileCopyrightText: 2021 RizinOrg <info@rizin.re>
+// SPDX-FileCopyrightText: 2021 deroad <wargio@libero.it>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_demangler.h>
+#include <rz_util.h>
+#include <rz_libdemangle.h>
+
+#include "demangler_plugins.c"
```

Comment:
let's avoid this inclusion of c files in new files if not strictly necessary. Just put the whole content here, I think it's fine.